### PR TITLE
feat: support publish new release image manually

### PR DESCRIPTION
**Pull request by keroro520**

### CODE REVIEW

File: .github/workflows/docker-publish-indexer.yml [hunk 1]
```diff
@@ -2,11 +2,8 @@ name: Docker Build Push (Indexer)
 
 on:
   push:
-    branches: [main, hot-fix, compatibility-breaking-changes, bump-pg-version]
-    # Publish semver tags as releases.
     tags: ["v*.*.*"]
-  pull_request:
-    branches: [main, hot-fix, compatibility-breaking-changes, bump-pg-version]
+  workflow_dispatch:
```
Comment (keroro520) [inline .github/workflows/docker-publish-indexer.yml:6]: Using `workflow_dispatch` trigger allows us to build specified images manually. I think it is better for us. We can build images by specifying a branch name.

**Comment by keroro520** [comment]
force-pushed [3a8a8bd ](https://github.com/nervosnetwork/godwoken-web3/commit/3a8a8bdbc08c8d335813babcc5425fc9790c204c)to [f18ac2c](https://github.com/nervosnetwork/godwoken-web3/commit/f18ac2c3a844fbabde1334a67403551612f3633e):

- add events trigger: `push:     branches: ["main", "rc*"]`

File: .github/workflows/docker-publish-indexer.yml [hunk 2]
```diff
@@ -2,11 +2,9 @@ name: Docker Build Push (Indexer)
 
 on:
   push:
-    branches: [main, hot-fix, compatibility-breaking-changes, bump-pg-version]
-    # Publish semver tags as releases.
+    branches: ["main", "rc*"]
```
[the remark this answers is not in the document]
Comment (RetricSu) [reply .github/workflows/docker-publish-indexer.yml:5]: the RC branch will be named to `MAJOR.MINOR-rc` (eg: [1.1-rc](https://github.com/nervosnetwork/godwoken-web3/tree/1.1-rc)), pls change the regex. @keroro520

**Comment by RetricSu** [comment]
btw, need to cherry-pick to 1.1-rc branch

File: .github/workflows/docker-publish-indexer.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (keroro520) [reply .github/workflows/docker-publish-indexer.yml:5]: What about `[0-9]+.[0-9]+-rc[0-9]*`?

Reference:
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

[the remark this answers is not in the document]
Comment (RetricSu) [reply .github/workflows/docker-publish-indexer.yml:5]: LGTM

[the remark this answers is not in the document]
Comment (keroro520) [reply .github/workflows/docker-publish-indexer.yml]: Updated it at [b1a9149](https://github.com/nervosnetwork/godwoken-web3/pull/366/commits/b1a9149c2805c5acad6bdadd7ee0588fc92bc4ab)

**Review by Flouse — APPROVED** [review]

File: .github/workflows/docker-publish-indexer.yml (hunk 1 above)
Comment (Flouse) [reply .github/workflows/docker-publish-indexer.yml:7]: Does workflow_dispatch require Admin role?

Comment (Flouse) [reply .github/workflows/docker-publish-indexer.yml:7]: I see.
> Write access to the repository is required to perform these steps.

**Pull request merged**